src/server/graphql/mutations/navigateMeeting.js

```diff
@@ -0,0 +1,68 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql';
+import getRethink from 'server/database/rethinkDriver';
+import {getUserId} from 'server/utils/authorization';
+import publish from 'server/utils/publish';
+import {TEAM} from 'universal/utils/constants';
+import {sendNotMeetingFacilitatorError} from 'server/utils/authorizationErrors';
+import NavigateMeetingPayload from 'server/graphql/types/NavigateMeetingPayload';
+import {sendMeetingNotFoundError, sendStageNotFoundError} from 'server/utils/docNotFoundErrors';
+import findStageById from 'universal/utils/meetings/findStageById';
+
+export default {
+  type: NavigateMeetingPayload,
+  description: 'update a meeting by marking an item complete and setting the facilitator location',
+  args: {
+    completedStageId: {
+      type: GraphQLID,
+      description: 'The stage that the facilitator would like to mark as complete'
+    },
+    facilitatorStageId: {
+      type: GraphQLID,
+      description: 'The stage where the facilitator is'
+    },
+    meetingId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The meeting ID'
+    }
+  },
+  async resolve(source,
+    {completedStageId, facilitatorStageId, meetingId},
+    {authToken, socketId: mutatorId, dataLoader}) {
+    const r = getRethink();
+    const now = new Date();
+    const operationId = dataLoader.share();
+    const subOptions = {mutatorId, operationId};
+
+    // AUTH
+    const viewerId = getUserId(authToken);
+    const meeting = await r.table('NewMeeting').get(meetingId).default(null);
+    if (!meeting) return sendMeetingNotFoundError(authToken, meetingId);
+    const {facilitatorUserId, phases, teamId} = meeting;
+    if (viewerId !== facilitatorUserId) return sendNotMeetingFacilitatorError(authToken, viewerId);
+
+    // VALIDATION
+    if (completedStageId) {
+      const completedStageRes = findStageById(phases, completedStageId);
+      if (!completedStageRes) return sendStageNotFoundError(authToken, completedStageId);
+      const {stage} = completedStageRes;
+      // MUTATIVE
```

Comment:
When this is commented as MUTATIVE, is this because stage is a object held within phases and phases is written back in the update below?

Reply:
_almost_. i just use that comment flag any time i mutate an object, regardless of whether i write it back to a database. i'm just saying "hey look, if you use `meeting` it's not the same object that it once was." in this case, it is SO much prettier than the immutable way.

Reply:
got it